data-collection/data-collection/Rich Popup/Related Records/ManyToOneRelationship.swift

```diff
@@ -72,3 +72,12 @@ class ManyToOneRelationship: Relationship {
         }
     }
 }
+
+// MARK: Errors
+
+extension ManyToOneRelationship {
+    struct RequiredRelationship: LocalizedError {
+        let name: String
+        var errorDescription: String? { "Missing required relationship named \"\(name)\"." }
```

Comment:
A non-optional property can witness for an optional one, which I think would be more efficient. Maybe not? I don't know.

```suggestion
        var errorDescription: String { "Missing required relationship named \"\(name)\"." }
```

If you wanted to, you could use avoid needing to escape the quotation marks using extended string delimiters:

```suggestion
        var errorDescription: String { #"Missing required relationship named "\#(name)"."# }
```